src/main/resources/avro/references.avdl

```diff
@@ -27,8 +105,22 @@ record Reference {
   string name;
 
   /**
+  Start and end locations of how this reference fits into the existing graph.
+  May be null only for a primary sequence that is not joined onto any other.
+  Note that the referenceId for start and end may not be the same.
+  */
+  union { null, ReferenceJoinLocation } startJoin, endJoin;
+
+  /**
+  The sequence to insert between startJoin and endJoin. Must be set if sourceURI
+  is null.
+  */
+  union { null, string } sequence = null;
```

Comment:
Sorry, I kind of forgot about that method.

It seems like the Right Thing to do would be to have `sourceURI` retain its current information-only semantics. Clients should not ever go download that FASTA in the normal course of using the API. (Should we put a comment on it to that effect?)

I don't think adding a new `sequence` field is a good idea either, because we want one correct way to get the sequence (through the method call). I think extending the method to take a Path as input might be a good idea, as it would push some of the IO-intensive walking of the graph from the client to the server, but it would be a bit hard to square with keeping the method as a GET request.

So I'll just knock off the `sequence` field and note that the sequences of `Reference`s, even tiny 1-base ones used to specify SNPs, should be obtained by calls to `ListReferenceBases`?